src/xdocs/checks/annotation/annotationlocation.xml.template

```diff
@@ -0,0 +1,230 @@
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#METHOD_DEF">
+                    METHOD_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CTOR_DEF">
+                    CTOR_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#VARIABLE_DEF">
+                    VARIABLE_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_DEF">
+                    ANNOTATION_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_FIELD_DEF">
+                    ANNOTATION_FIELD_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#RECORD_DEF">
+                    RECORD_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#COMPACT_CTOR_DEF">
+                    COMPACT_CTOR_DEF</a>
+                  .
+              </td>
+              <td>
+                <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CLASS_DEF">
+                    CLASS_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INTERFACE_DEF">
+                    INTERFACE_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#PACKAGE_DEF">
+                    PACKAGE_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_CONSTANT_DEF">
+                    ENUM_CONSTANT_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF">
+                    ENUM_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#METHOD_DEF">
+                    METHOD_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CTOR_DEF">
+                    CTOR_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#VARIABLE_DEF">
+                    VARIABLE_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#RECORD_DEF">
+                    RECORD_DEF</a>
+                , <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#COMPACT_CTOR_DEF">
+                    COMPACT_CTOR_DEF</a>
+                  .
+              </td>
+              <td>6.0</td>
+            </tr>
+          </table>
+        </div>
+      </subsection>
+
+      <subsection name="Examples" id="Examples">
+        <p id="Example1-config">
+          To configure the default check to allow one single parameterless annotation on the same
+            line:
+        </p>
+        <source>
+${Example1-config}
+        </source>
```

Comment:
I am unsure about making those templates with examples now. If a change to some xdoc has to be made, it has to be made here too. 

What we can do is add those `id="Example1.config"` anchors and later generate the template from them. Just pick the `source` after the `p` and swap the contents with the value of the `id`.

Reply:
> I am unsure about making those templates with examples now. If a change to some xdoc has to be made, it has to be made here too.

Please provide some details or an example, I am not sure exactly what this means.

Reply:
What I mean is, it will take a month before we start generating xdocs from templates. If an example is added or documentation is written in that time(1 month), both the template and the original file have to be changed.

Reply:
original file like https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/checks/annotation/annotationlocation.xml stays in git.
we create another file that is in template.
each time `mvn test` are executing, there some Test class, that actually not a test but just executes by junit, this "Test" generates src/xdocs/checks/annotation/annotationlocation.xml from template, override file that is in git. This "Test" does not touch any other xdoc files if template is not present.

if somebody will update xdoc and forget to update template, it will be visible in git diff, and be blocked in PR review.

this will help us module by module migrate to templates (just new files are created). No blockers to other, release at any time, invisible to others, any regression on examples parser or ..... will be detected immediately by unexpected git diff.

as all are migrated and we have time and we really want to remove static xdoc modules files.
We can start final phase is to figure out how to remove xdoc static files and play with site phase to generate files to special folder and site phase will take files from generated area and combine with static xdoc (not for modules) pages. There are a lot of unknown here for me on how to do this. I have no idea who to do this to be honest, but it might be possible.

Reply:
On a moment of finishing template for module and having all examples in separate files, we need to remove this module from validation of javadoc-xdoc to have same content.